src/openms/include/OpenMS/FILTERING/CALIBRATION/PrecursorCorrection.h

```diff
@@ -0,0 +1,163 @@
+
+
+    /**
+    @brief Writer can be used in association with correctToNearestMS1Peak or correctToHighestIntensityMS1Peak
+
+    Format:
+    RT	    uncorrectedMZ	correctedMZ	deltaMZ
+    100.1	509.9999	    510	         0.0001
+    180.9	610.0001	    610	        -0.0001
+    183.92	611.0035	    611.0033	  -0.0002
+    */
+    static void writeHist(const String& out_csv,
+                          const std::vector<double> & delta_mzs,
+                          const std::vector<double> & mzs,
+                          const std::vector<double> & rts);
+    /**
+    @brief Selection of the peak in closest proximity as corrected precursor mass in a given mass range (e.g. precursor mass +/- 0.2 Da)
+
+    For each MS2 spectrum the corresponding MS1 spectrum is determined by using the rt information of the precursor.
+    In the MS1, the peak closest to the uncorrected precursor m/z is selected and used as corrected precursor m/z.
+    */
+    static std::set<Size> correctToNearestMS1Peak(MSExperiment & exp,
+                                                  double mz_tolerance,
+                                                  bool ppm,
+                                                  std::vector<double> & delta_mzs,
+                                                  std::vector<double> & mzs,
+                                                  std::vector<double> & rts);
+
+     /**
+     @brief Selection of the peak with the highest intensity as corrected precursor mass in a given mass range (e.g. precursor mass +/- 0.2 Da)
+
+     For each MS2 spectrum the corresponding MS1 spectrum is determined by using the rt information of the precursor.
+     In the MS1, the peak with the highest intensity in a given mass range to the uncorrected precursor m/z is selected and used as corrected precursor m/z.
+     */
+     static std::set<Size> correctToHighestIntensityMS1Peak(MSExperiment & exp,
+                                                           double mz_tolerance,
+                                                           std::vector<double> & delta_mzs,
+                                                           std::vector<double> & mzs,
+                                                           std::vector<double> & rts);
+
+
+    /**
+    @brief Reassigns a precursor to the nearest feature in a given rt and mass range.
+    Wrong assignment of the mono-isotopic mass for precursors are assumed:
+    - if precursor_mz matches the mz of a non-monoisotopic feature mass trace
+    - and in the case that believe_charge is true: if feature_charge matches the precursor_charge
+    In the case of wrong mono-isotopic assignment several options for correction are available:
+    keep_original will create a copy of the precursor and tandem spectrum for the new mono-isotopic mass trace and retain the original one
+    all_matching_features does this not for only the closest feature but all features in a question
```

Comment:
can you use the @param option?

Reply:
Ok, I will use the param option.

Reply:
@oliveralka I suggest to adhere to this https://github.com/OpenMS/OpenMS/wiki/Coding-conventions#doxygen